src/toolbar/Toolbar.cpp

```diff
@@ -0,0 +1,233 @@
+            ImGui::SameLine();
+            if (ImGui::Button("Console"))
+                selectedID = ToolbarWidgetID::CONSOLE;
+            ImGui::SameLine();
+            if (ImGui::Button("Keybinds"))
+                selectedID = ToolbarWidgetID::KEYBINDS;
+            ImGui::SameLine();
+            if (ImGui::Button("Settings"))
+                selectedID = ToolbarWidgetID::SETTINGS;
+
+            ImGui::Spacing();
+        }
+        if (selectedID < ToolbarWidgetID::COUNT)
+            SetActiveWidget(selectedID);
+        
+        int activeCount = 0;
+        ImVec2 zeroVec = {0, 0};
+        if (!Options::IsFirstLaunch)
+        {
+            if (m_activeWidgetID == ToolbarWidgetID::MODS)
+            {
+                if (ImGui::BeginChild("Mod widgets", zeroVec, true))
+                    m_mods.Update();
+                ImGui::EndChild();
+                ++activeCount;
+            }
+            if (m_activeWidgetID == ToolbarWidgetID::CONSOLE)
+            {
+                if (ImGui::BeginChild("Console", zeroVec, true))
+                    m_console.Update();
+                ImGui::EndChild();
+                ++activeCount;
+            }
+            if (m_activeWidgetID == ToolbarWidgetID::KEYBINDS)
+            {
+                if (ImGui::BeginChild("Keybinds", zeroVec, true))
+                    m_keybinds.Update();
+                ImGui::EndChild();
+                ++activeCount;
+            }
+        }
+        if (m_activeWidgetID == ToolbarWidgetID::SETTINGS)
+        {
+            if (ImGui::BeginChild("Settings", zeroVec, true))
+                m_settings.Update();
+            ImGui::EndChild();
+            ++activeCount;
+        }
+
+        if (activeCount == 0)
+            SetActiveWidget(ToolbarWidgetID::COUNT);
+    }
+    ImGui::End();
+}
+
+bool Toolbar::IsInitialized() const
+{
+    return m_initialized;
+}
+
+LRESULT Toolbar::OnWndProc(HWND, UINT auMsg, WPARAM awParam, LPARAM)
+{
+    // TODO - is this useful now?
+    return 0;
+}
+
+BOOL Toolbar::ClipToCenter(RED4ext::CGameEngine::UnkC0* apThis)
+{
+    HWND wnd = (HWND)apThis->hWnd;
+    HWND foreground = GetForegroundWindow();
+
+    if(wnd == foreground && apThis->unk164 && !apThis->unk140 && !Get().IsEnabled())
+    {
+        RECT rect;
+        GetClientRect(wnd, &rect);
+        ClientToScreen(wnd, reinterpret_cast<POINT*>(&rect.left));
+        ClientToScreen(wnd, reinterpret_cast<POINT*>(&rect.right));
+        rect.left = (rect.left + rect.right) / 2;
+        rect.right = rect.left;
+        rect.bottom = (rect.bottom + rect.top) / 2;
+        rect.top = rect.bottom;
+        apThis->isClipped = true;
+        ShowCursor(FALSE);
+        return ClipCursor(&rect);
+    }
+
+    if(apThis->isClipped)
+    {
+        apThis->isClipped = false;
+        return ClipCursor(nullptr);
+    }
+
+    return 1;
+}
+
+void Toolbar::Hook()
+{
+    uint8_t* pLocation = FindSignature({
```

Comment:
Why is the mouse clip in the toolbar?

Reply:
well, hopefully better now... 

dont know if you want to completely separate that widget stuff to its own class from this and let overlay own just that class...
or if more-or-less main issue was that it was called toolbar when toolbar was kind of small part of it